app/steps/medicaid/amounts_income.rb

```diff
@@ -14,36 +18,53 @@ class AmountsIncome < MemberStep
       to: :member,
     )
 
-    def valid?
-      if jobs_and_income_dont_match?
-        errors.add(:incomes_missing, "Make sure all jobs have an income amount")
-        false
-      elsif unemployment_income_is_not_provided?
-        errors.add(:incomes_missing, "Make sure to provide unemployment income")
-        false
-      elsif self_employment_income_not_provided?
-        errors.add(
-          :incomes_missing,
-          "Make sure to provide self-employment income",
-        )
-        false
-      else
-        true
+    validate :incomes
+    validate :employer_names
+    validate :payment_frequencies
+    validate :unemployment_income_provided
+    validate :self_employment_income
+
+    private
+
+    def incomes
+      return true if employed_number_of_jobs == employed_monthly_income&.count
+      errors.add(:incomes_missing, "Make sure all jobs have an income amount")
+    end
+
+    def employer_names
+      return true if employed_number_of_jobs == employed_monthly_employer&.count
+      errors.add(:incomes_missing, "Make sure all jobs have an employer name")
+    end
+
+    def payment_frequencies
+      if employed_number_of_jobs == employed_payment_frequency&.count
+        return true
       end
+
+      errors.add(
+        :incomes_missing,
+        "Make sure all jobs have a payment frequency",
+      )
     end
 
-    private
+    def unemployment_income_provided
+      return true if unemployed_and_unemployment_income_provided?
+      errors.add(:incomes_missing, "Make sure to provide unemployment income")
+    end
 
-    def jobs_and_income_dont_match?
-      employed_number_of_jobs != employed_monthly_income&.count
+    def unemployed_and_unemployment_income_provided?
+      !receives_unemployment_income? ||
+        (receives_unemployment_income? && unemployment_income.present?)
     end
 
-    def unemployment_income_is_not_provided?
-      receives_unemployment_income? && unemployment_income.blank?
+    def self_employment_income
+      return true if self_employment_income_provided?
+      errors.add(:incomes_missing, "Make sure to provide unemployment income")
     end
 
-    def self_employment_income_not_provided?
-      self_employed? && self_employed_monthly_income.blank?
+    def self_employment_income_provided?
```

Comment:
same comment as above re: method will be simplified if we check for invalid states and then return an error rather than checking for a valid state and returning true